### PR TITLE
Feature/recommendations

### PR DESCRIPTION
**Pull request by ccruzt**
Add horizontal scroll with recommended movies on detail screen

https://user-images.githubusercontent.com/92454108/161325457-7cc0d6f1-e83a-4d8d-82ed-15314243c60c.mov

### CODE REVIEW

File: androidHelpers/src/main/java/com/example/androidHelpers/compose/views/MovieHorizontalItem.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+fun MovieHorizontalItem(thumbnailUrl: String, movieName : String, modifier: Modifier = Modifier) {
+    Card(
+        modifier = modifier
+            .width(150.dp)
+            .padding(5.dp)
+            .height(180.dp),
+        shape = RoundedCornerShape(10.dp),
+        elevation = 5.dp,
+    ) {
+        Box(modifier = Modifier.height(250.dp)) {
```
Comment (ManuelUrbanoB) [inline androidHelpers/src/main/java/com/example/androidHelpers/compose/views/MovieHorizontalItem.kt]: please, add this to the resources

File: androidHelpers/src/main/java/com/example/androidHelpers/compose/views/MovieHorizontalItem.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                        )
+                    )
+            )
+            Box(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(12.dp),
+                contentAlignment = Alignment.BottomStart
+            ) {
+                Text(movieName, style = TextStyle(color = Color.White, fontSize = 16.sp))
```
Comment (ManuelUrbanoB) [inline androidHelpers/src/main/java/com/example/androidHelpers/compose/views/MovieHorizontalItem.kt]: same here

File: movies/presentation/src/main/java/com/example/movies/presentation/ui/movieDetail/MovieDetailFragment.kt [hunk 3]
```diff
@@ -41,28 +42,44 @@ class MovieDetailFragment : BaseFragment() {
...
+                    }
+                }
+            }
+        }
+        lifecycleScope.launch {
+            viewModel.movieRecommendationUiState.collect { uiState ->
+                when (uiState) {
+                    is MovieRecommendationUiState.Init -> isLoadingVisible.value = false
+                    is MovieRecommendationUiState.Loading -> isLoadingVisible.value = true
+                    is MovieRecommendationUiState.GetMovieRecommendations -> {
```
Comment (ManuelUrbanoB) [inline movies/presentation/src/main/java/com/example/movies/presentation/ui/movieDetail/MovieDetailFragment.kt]: this line should be MovieRecommendationUiState.GetMovieRecommendations ->                      setSuccessMovieRecommendations(uiState)

File: movies/data/src/main/java/com/example/movies/data/repositories/MoviesRepositoryImpl.kt [hunk 4]
```diff
@@ -76,4 +78,18 @@ class MoviesRepositoryImpl(
             Response.Error(error.message.toString())
         }
     }
+
+    override suspend fun getRecommendedMovies(id: Int): Response<MovieRecommendationList> {
```
Comment (ManuelUrbanoB) [inline movies/data/src/main/java/com/example/movies/data/repositories/MoviesRepositoryImpl.kt]: please, add page in this function  and in the use case send one 1 to the page

File: movies/domain/src/main/java/com/example/movies/domain/repositories/MoviesRepository.kt [hunk 5]
```diff
@@ -3,16 +3,16 @@ package com.example.movies.domain.repositories
...
-import kotlinx.coroutines.flow.Flow
 import com.example.movies.domain.entities.MovieList
+import com.example.movies.domain.entities.MovieRecommendationList
+import kotlinx.coroutines.flow.Flow
 
 interface MoviesRepository {
     suspend fun getListMovies(page: Int?): Response<MovieList>
     suspend fun getDetailMovie(id: Int): Response<MovieDetail>
-
+    suspend fun getRecommendedMovies(id: Int): Response<MovieRecommendationList>
```
Comment (ManuelUrbanoB) [inline movies/domain/src/main/java/com/example/movies/domain/repositories/MoviesRepository.kt]: change the interface to can implement the repository with the page in recomendations.

**Review by ManuelUrbanoB — CHANGES_REQUESTED** [review]
MovieDetailMapper and MovieMapper are empty, review those files.

**Review by ManuelUrbanoGlobant — APPROVED** [review]

**Pull request merged**